pyproject.toml

```diff
@@ -145,7 +145,7 @@ known-third-party = [
 
 # The Pydantic representations of SqlAlchemy Models are not parsed well with Pydantic
 # when __future__.annotations is used so we need to skip them from upgrading
-"airflow/serialization/pydantic/*.py" = ["I002"]
+"airflow/serialization/pydantic/*.py" = ["I002", "UP007"]
```

Comment:
What requires this ignore rule?

Reply:
Otherwise Pydantic models (in `airflow/serialization/pydantic`) are using the syntax `... | None` instead of `Optional[...]`. Despite the fact we are using the syntax `... | None` across the code, Pydantic is not compatible with it. We need to make an exception then